vespalib/src/vespa/vespalib/util/memory_trap.cpp

```diff
@@ -0,0 +1,171 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#ifdef __linux__
+#  define VESPA_HAS_MPROTECT
+#endif
+
+#include "memory_trap.h"
+#include <string_view>
+#include <cassert>
+#include <cerrno>
+#include <cstdint>
+#include <cstdlib>
+#include <cstring>
+#include <malloc.h>
+#ifdef VESPA_HAS_MPROTECT
+#  include <unistd.h>
+#  include <sys/mman.h>
+#endif
+
+#include <vespa/log/log.h>
+LOG_SETUP(".vespalib.util.memory_trap");
+
+using namespace std::string_view_literals;
+
+namespace vespalib {
+
+namespace {
+
+#ifdef VESPA_HAS_MPROTECT
```

Comment:
I am not a fan of #ifdefs. Is it possible to gather all 5 into just a single one. That makes it far more readable and maintainable.
You get clean method definitions. One dummy set and one for mprotect.

Reply:
Fixed